clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -183,6 +183,10 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
         return new NetworkClientDelegate.PollResult(heartbeatRequestState.heartbeatIntervalMs, Collections.singletonList(request));
     }
 
+    public MembershipManager membershipManager() {
+        return membershipManager;
+    }
```

Comment:
Why do we need this?

Reply:
It's for access by the `ApplicationEventProcessor` to process its events that relate to the state machine.